src/Neo/SmartContract/Manifest/ContractManifest.cs

```diff
@@ -110,22 +110,27 @@ public StackItem ToStackItem(ReferenceCounter referenceCounter)
         /// <returns>The converted manifest.</returns>
         public static ContractManifest FromJson(JObject json)
         {
-            ContractManifest manifest = new()
+            ContractManifest manifest = new();
+            try
             {
-                Name = json["name"].GetString(),
-                Groups = ((JArray)json["groups"]).Select(u => ContractGroup.FromJson((JObject)u)).ToArray(),
-                SupportedStandards = ((JArray)json["supportedstandards"]).Select(u => u.GetString()).ToArray(),
-                Abi = ContractAbi.FromJson((JObject)json["abi"]),
-                Permissions = ((JArray)json["permissions"]).Select(u => ContractPermission.FromJson((JObject)u)).ToArray(),
-                Trusts = WildcardContainer<ContractPermissionDescriptor>.FromJson(json["trusts"], u => ContractPermissionDescriptor.FromJson((JString)u)),
-                Extra = (JObject)json["extra"]
-            };
+                manifest.Name = json["name"]?.GetString();
```

Comment:
previously fail when null, now set name as null

Reply:
previously crash when null, now it throws a FormatException.

Reply:
previously crash when null, now it throws a FormatException. Demo:
![image](https://github.com/neo-project/neo/assets/6295602/ec0d2abf-a12b-468f-9a43-b000e88b552a)

Reply:
`name` is required. 

https://github.com/neo-project/proposals/blob/master/nep-15.mediawiki#user-content-Manifest

Reply:
but this maybe can produce a fork

Reply:
> previously crash when null, now it throws a FormatException. Demo:
> ![image](https://github.com/neo-project/neo/assets/6295602/ec0d2abf-a12b-468f-9a43-b000e88b552a)
> 

a big no here. null and exception are totally different behavior.

Reply:
I just fixed the bug where the wrong json was causing a crash. Previously the node would simply crash without throwing an exception.

Reply:
> `name` is required.
> 
> https://github.com/neo-project/proposals/blob/master/nep-15.mediawiki#user-content-Manifest

Yes, manifest.Name is not allowed to be null, there's another line of code after it (it was there before, not a new one I added)
So it won't cause a fork @shargon @cschuchardt88 
```
if (string.IsNullOrEmpty(manifest.Name))
    throw new FormatException();
```

Reply:
then use `!` instead of `?` to avoid the behavior change

Reply:
fixed